File: java/src/main/java/com/mstgnz/blog/entities/Blog.java
package com.mstgnz.blog.entities;

import lombok.*;

import javax.persistence.*;
import java.io.Serializable;
import java.util.Date;
import java.util.List;

@Entity
@Table(name = "blogs")
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@EqualsAndHashCode(of = {"id"})
@ToString
public class Blog implements Serializable, IEntity {
    @Id
    @SequenceGenerator(name="seq_blog", allocationSize = 1)
    @GeneratedValue(generator = "seq_blog", strategy = GenerationType.SEQUENCE)
    private Long id;
    private Long userId;
    @Column(length = 100)
    private String title;
    @Column(length = 150)
    private String slug;
    @Column(length = 255)
    private String shortText;
    @Column(length = 10000)
    private String longText;
    private Boolean status;
    @Temporal(TemporalType.TIMESTAMP)
    @Column(columnDefinition = "TIMESTAMP WITHOUT TIME ZONE")
    private Date createDate;
    @Temporal(TemporalType.TIMESTAMP)
    @Column(updatable = false,
            columnDefinition = "TIMESTAMP WITHOUT TIME ZONE")
    private Date updateDate;
}

File: go/server/dto/blog.go
package dto

type BlogCreateDTO struct {
	UserID    uint64 `json:"user_id" form:"user_id"`
	Title     string `json:"title" form:"title" binding:"required"`
	ShortText string `json:"short_text" form:"short_text" binding:"required"`
	LongText  string `json:"long_text" form:"long_text" binding:"required"`
}

type BlogUpdateDTO struct {
	ID        uint64 `json:"id" form:"id" binding:"required"`
	UserID    uint64 `json:"user_id" form:"user_id"`
	Title     string `json:"title" form:"title" binding:"required"`
	ShortText string `json:"short_text" form:"short_text" binding:"required"`
	LongText  string `json:"long_text" form:"long_text" binding:"required"`
}

type BlogListDTO struct {
	ID         uint64 `json:"id"`
	UserID     uint64 `json:"userID"`
	Title      string `json:"title"`
	Slug       string `json:"slug"`
	ShortText  string `json:"shortText"`
	LongText   string `json:"longText"`
	CreateDate string `json:"createDate"`
	UpdateDate string `json:"updateDate"`
}
